homework01/artur/vector.h

```diff
@@ -0,0 +1,25 @@
+#include <iostream>
+
+class Vector
+{
+ public:
+  Vector();
+  Vector(const int, const int);
+  Vector(const Vector&);
+  
+  const int GetX() const;
+  const int GetY() const;
+
+  const bool operator==(const Vector&);
+  const Vector operator+(const Vector&);
+
+  Vector& operator+=(const Vector&);
+  
+  void Add(const Vector&);
+
+  friend std::ostream& operator<<(std::ostream&, const Vector&);
```

Comment:
Is friendship needed here?

Reply:
It is if I want to declare the overload inside my class, right? I can skip it if the declaration is outside it, but I think that such overload is cleaner inside the class.

Reply:
In this form of `operator<<` (first parameter of type other than `Vector`), it is a non-member function. A non-member function can access any of Vector's public methods. If you are using `GetX` and `GetY` in this `operator<<`, then it is fine, it does not need to be friend with `Vector`. Friendship should only be used when it is needed to access private or protected members.